components/select/nz-select-top-control.component.html

```diff
@@ -34,21 +39,28 @@
   </ng-container>
   <!--multiple or tags mode-->
   <ul *ngIf="nzSelectService.isMultipleOrTags">
-    <ng-container *ngFor="let option of nzSelectService.listOfCachedSelectedOption | slice: 0 : nzMaxTagCount;trackBy:trackValue;">
-      <li [@zoomMotion]
-        [nzNoAnimation]="noAnimation?.nzNoAnimation"
-        [attr.title]="option.nzLabel"
-        [class.ant-select-selection__choice__disabled]="option.nzDisabled"
-        class="ant-select-selection__choice">
-        <div class="ant-select-selection__choice__content">{{ option.nzLabel }}</div>
-        <span *ngIf="!option.nzDisabled"
-          class="ant-select-selection__choice__remove"
-          (mousedown)="$event.preventDefault()"
-          (click)="removeSelectedValue(option, $event)">
-          <i nz-icon type="close" class="ant-select-remove-icon" *ngIf="!nzRemoveIcon; else nzRemoveIcon"></i>
-        </span>
-      </li>
+    <ng-container *ngIf="nzCustomTemplate; else noTemplate">
+      <ng-container *ngIf="nzSelectService.listOfCachedSelectedOption.length">
+        <ng-template [ngTemplateOutlet]="nzCustomTemplate" [ngTemplateOutletContext]="{ $implicit: nzSelectService.listOfCachedSelectedOption }"></ng-template>
```

Comment:
this template should set inside the `*ngFor`

Reply:
This is not the custom template for the options, it's just for the top selected label, so I don't think it should be inside the `*ngFor`, unless I'm missing something.

Reply:
the template should replace each selected option template other than all the template.
```
@Input() nzCustomTemplate: TemplateRef<{ $implicit: NzOptionComponent }>
```

Reply:
Yes I understand this one, but not sure I get your point of moving it into `*ngFor`

Reply:
```html
      <li [@zoomMotion]
        [nzNoAnimation]="noAnimation?.nzNoAnimation"
        [attr.title]="option.nzLabel"
        [class.ant-select-selection__choice__disabled]="option.nzDisabled"
        class="ant-select-selection__choice">
        <div class="ant-select-selection__choice__content">
                  <ng-template [ngTemplateOutlet]="nzCustomTemplate" [ngTemplateOutletContext]="{ $implicit: nzSelectService.listOfCachedSelectedOption }"></ng-template>
        </div>
        <span *ngIf="!option.nzDisabled"
          class="ant-select-selection__choice__remove"
          (mousedown)="$event.preventDefault()"
          (click)="removeSelectedValue(option, $event)">
          <i nz-icon type="close" class="ant-select-remove-icon" *ngIf="!nzRemoveIcon; else nzRemoveIcon"></i>
        </span>
      </li>
```

Reply:
Ah right I see what you mean now, haven't seen this code for months I totally forgot what I did... it make perfect sense I'll make a change now.

Reply:
great